packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,633 @@
+
+### shape-count
+
+`shape-count` should be a component that only subscribes and reads from the store. Let us create a custom element and render the table:
+
+```ts
+import { LitElement, html} from 'lit';
+import { customElement} from 'lit/decorators.js';
+
+@customElement('shape-count')
+export class ShapeCount extends LitElement {
+  render() {
+    return html`
+      <table>
+        <tr>
+          <td>${0}</td>
+          <td>circles</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>squares</td>
+        </tr>
+        <tr>
+          <td>${0}</td>
+          <td>total</td>
+        </tr>
+      </table>
+    `;
+  }
+}
+```
+
+Next we want to render actual counts rather than just `0`. In this case we will need all values from the state:
+
+- `state.circles`
+- `state.squares`
+- `state.shapeList`
+
+To achieve this, we will need a broad Redux selector that selects the entire state:
+
+```ts
+...
+import { SelectorController } from './selector-controller.js';
+import { store } from './store.js';
+
+@customElement('shape-count')
+export class ShapeCount extends LitElement {
+  private sc = new SelectorController(this, store, (state) => state);
+
+  render() {
+    const {circles, squares, shapeList} = this.sc.value;
+
+    return html`
+      <table>
+        <tr>
+          <td>${circles}</td>
+          <td>circles</td>
+        </tr>
+        <tr>
+          <td>${squares}</td>
+          <td>squares</td>
+        </tr>
+        <tr>
+          <td>${shapeList.length}</td>
+          <td>total</td>
+        </tr>
+      </table>
+    `;
+  }
+}
+```
+
+The counts are now pulled from the state in Redux and any updates to the state will update the component!
+
+To accomplish this, we initialized the `SelectorController` with the shared Redux store and rendered the entire state.
+
+### shape-list
+
+`shape-list` should be a component that only subscribes and reads the `state.shapeList` state from the store. Let's create a custom element with boilerplate, and render an array of `<div>` elements with classes set to the shape name. Our pre-provided CSS styles will render the squares and circles based on the class name.
+
+```ts
+import { customElement } from 'lit/decorators.js';
+import { LitElement, html} from 'lit';
+
+@customElement('shape-list')
+export class ShapeList extends LitElement {
+  render() {
+    const shapeList = [];
+
+    return shapeList.map(
+        shape => html`<div class="${shape}"></div>`
+    );
+  }
+}
+```
+
+Next, let’s initialize our `SelectorController` to hook into Redux and render the `shapeList`:
```

Comment:
Worth calling out that this time the selector is focussing a specific value (`shapeList`), instead of selecting the whole store with an identity function?

The difference here is that this component will only re-render if specifically `shapeList` changes.